From bfbe97c8b34f825b0d6b8ce5119bbcb7c892a55d Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Wed, 26 May 2021 21:22:24 +0300
Subject: [PATCH] meson: force the gir target to depend on wp-gtkdoc.h
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

With a hacky workaround, because the generate_gir() implementation
has a bug and does not pass the file dependencies properly down
to the custom_target() that it creates internally
---
 docs/meson.build | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/docs/meson.build b/docs/meson.build
index 08ae963..9c392b4 100644
--- a/docs/meson.build
+++ b/docs/meson.build
@@ -106,8 +106,25 @@ if build_gir
     build_by_default: true,
   )
 
+  # A dummy library dependency to force meson to make the gir target
+  # depend on wp-gtkdoc.h, because generate_gir() doesn't add dependencies
+  # on its sources (meson bug)
+  dummy_c = custom_target('dummy_c',
+    command: ['echo', 'int dummy(void) { return 0; }'],
+    capture: true,
+    output: 'dummy.c',
+    build_by_default: true,
+  )
+  libdummy = library('dummy',
+    wp_gtkdoc_h, dummy_c,
+    install: false,
+    soversion: '0',
+    version: '0.0.0',
+  )
+  dummy_dep = declare_dependency(link_with: libdummy)
+
   gnome.generate_gir(wp_lib,
-    dependencies: wp_dep,
+    dependencies: [wp_dep, dummy_dep],
     namespace: 'Wp',
     nsversion: wireplumber_api_version,
     sources: [wpenums_h, wp_lib_parse_headers, wp_gtkdoc_h],
-- 
2.25.1

